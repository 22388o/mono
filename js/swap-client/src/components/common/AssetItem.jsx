import React from 'react'
// mui import
import { Grid, Stack, Typography } from '@mui/material'
import NavigateNextIcon from '@mui/icons-material/NavigateNext'

export const AssetItem = ({ asset, handleClick }) => {
  return (
    <Stack 
      direction='row' 
      onClick={e => handleClick(asset)} style={{ cursor: 'pointer' }}
      className='asset-item'
      sx={{ justifyContent: 'space-between' }}
    >
      <Stack direction='row' spacing={0.5} sx={{alignItems:'center'}}>
        <img width={32} height={32} src={asset.img_url} />
        <Stack>
          <Typography>{ asset.short }</Typography>
          {asset.type && <h5 style={{ fontSize: '0.8em', color: 'grey' }}>{asset.type}</h5>}
        </Stack>
      </Stack>
      <Stack direction='row' alignItems='center' gap={0.5}>
        { asset.amount }
        <span style={{ color: 'grey', fontSize: '0.7em' }}>{asset.type && asset.type.toUpperCase()}</span>
        <NavigateNextIcon />
      </Stack>
    </Stack>
  )
}
