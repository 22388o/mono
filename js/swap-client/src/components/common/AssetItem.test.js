import { AssetItem } from './AssetItem';
import { create, act } from 'react-test-renderer';

describe('AccessOptionComponent', () => {
  it('should work', () => {
    let tree;
    let asset = {
      img_url: '123',
      title: 'card',
      type: '123',
      amount: 1
    }
    let handleClick = jest.fn();

    act(() => {
      tree = create(
        <AssetItem 
          asset={asset} 
          handleClick={handleClick} />
      );
    });

    expect(tree).toMatchSnapshot();
  });
  afterAll(() => jest.resetModules());
});