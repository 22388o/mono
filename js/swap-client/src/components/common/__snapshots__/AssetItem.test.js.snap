// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AccessOptionComponent should work 1`] = `
<div
  className="MuiStack-root asset-item css-gmwslw-MuiStack-root"
  onClick={[Function]}
  style={
    {
      "cursor": "pointer",
    }
  }
>
  <div
    className="MuiStack-root css-13w4gzo-MuiStack-root"
  >
    <img
      height={32}
      src="123"
      width={32}
    />
    <div
      className="MuiStack-root css-nen11g-MuiStack-root"
    >
      <p
        className="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
      />
      <h5
        style={
          {
            "color": "grey",
            "fontSize": "0.8em",
          }
        }
      >
        123
      </h5>
    </div>
  </div>
  <div
    className="MuiStack-root css-1jtdgec-MuiStack-root"
  >
    1
    <span
      style={
        {
          "color": "grey",
          "fontSize": "0.7em",
        }
      }
    >
      123
    </span>
    <svg
      aria-hidden={true}
      className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
      data-testid="NavigateNextIcon"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M10 6 8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z"
      />
    </svg>
  </div>
</div>
`;
